kafka-admin-client/src/test/java/io/mosip/kafkaadminclient/MosipKafkaAdminClientTest.java

```diff
@@ -0,0 +1,100 @@
+package io.mosip.kafkaadminclient;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.CreateTopicsResult;
+import org.apache.kafka.clients.admin.DescribeTopicsResult;
+import org.apache.kafka.clients.admin.KafkaAdminClient;
+import org.apache.kafka.clients.admin.ListTopicsResult;
+import org.apache.kafka.clients.admin.TopicDescription;
+import org.apache.kafka.common.KafkaFuture;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ Admin.class })
+public class MosipKafkaAdminClientTest {
+
+    @SuppressWarnings("unchecked")
+	@Test
+	public void isTopicsPresentTest() throws Exception {
+		PowerMockito.mockStatic(Admin.class);
+		KafkaAdminClient adminMock = mock(KafkaAdminClient.class);
+		PowerMockito.when(Admin.create(any(Properties.class))).thenReturn(adminMock);
+    	ListTopicsResult listTopicsResult = Mockito.mock(ListTopicsResult.class);
+		KafkaFuture<Set<String>> kafkaFuture = Mockito.mock(KafkaFuture.class);
+		Set<String> sampleStrings = new HashSet<>();
+		sampleStrings.add("test");
+		Mockito.when(kafkaFuture.get()).thenReturn(sampleStrings);
+		Mockito.when(listTopicsResult.names()).thenReturn(kafkaFuture);
+		Mockito.when(adminMock.listTopics(Mockito.any())).thenReturn(listTopicsResult);
+		MosipKafkaAdminClient mosipKafkaAdminClient = new MosipKafkaAdminClient("test");
+		Assert.assertTrue(mosipKafkaAdminClient.isTopicsPresent("test"));
```

Comment:
Can we have one more test where we check for a topic that is not present in the mock list ?

Reply:
Can we have more test case where the topics is given with comma separated ? Also have a case with both present, one present and none present etc...

Reply:
Done.